gbif-occurrence-linkage/main.py

```diff
@@ -2,102 +2,70 @@
+        value_serializer=lambda m: json.dumps(m).encode('utf-8'))
     for msg in consumer:
         try:
             logging.info('Received message: ' + str(msg.value))
             json_value = msg.value
-            specimen_data = json_value['object']['digitalSpecimen']
+            shared.mark_job_as_running(json_value["jobId"])
+            specimen_data = json_value['object']
             result = run_api_call(specimen_data)
-            annotations = map_to_annotation(specimen_data, result, json_value["jobId"])
-            send_updated_opends(annotations, producer)
+            annotation_event = map_to_annotation_event(specimen_data, result,
+                                                  json_value["jobId"])
+            publish_annotation_event(annotation_event, producer)
         except Exception as e:
             logging.exception(e)
 
 
-def map_to_annotation(specimen_data: Dict, result: Dict[str, str], job_id: str) -> dict:
+def map_to_annotation_event(specimen_data: Dict, result: Dict[str, str],
+        job_id: str) -> dict:
     """
     Map the result of the API call to an annotation
     :param specimen_data: The JSON value of the Digital Specimen
     :param result: The result which contains either the GBIF ID or an error message
     :param job_id: The job ID of the MAS
     :return: Returns a formatted annotation Record which includes the Job ID
     """
-    timestamp = timestamp_now()
-    oa_value = {
-        'entityRelationships': {
-            'entityRelationshipType': 'hasGbifID',
-            'objectEntityIri': f'https://www.gbif.org/occurrence/{result["gbifId"]}',
-            'entityRelationshipDate': timestamp,
-            'entityRelationshipCreatorName': os.environ.get('MAS_NAME'),
-            'entityRelationshipCreatorId': f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        }
-    }
-    annotation = {
-        'rdf:type': 'Annotation',
-        'oa:motivation': 'ods:adding',
-        'oa:creator': {
-            ODS_TYPE: 'oa:SoftwareAgent',
-            'foaf:name': os.environ.get('MAS_NAME'),
-            ODS_ID: f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        },
-        'dcterms:created': timestamp,
-        'oa:target': {
-            ODS_ID: specimen_data[ODS_ID],
-            ODS_TYPE: specimen_data[ODS_TYPE],
-            'oa:selector': {
-                ODS_TYPE: 'ClassSelector',
-                'oa:class': '$.entityRelationships'
-            },
-        },
-        'oa:body': {
-            ODS_TYPE: 'TextualBody',
-            'oa:value': [json.dumps(oa_value)],
-            'dcterms:reference': result['queryString']
-        }
-    }
-    mas_job_record = {
-        "jobId": job_id,
-        "annotations": [annotation]
+    timestamp = shared.timestamp_now()
+    ods_agent = shared.get_agent()
+    oa_value = shared.map_to_entity_relationship('hasGbifID',
+                                          f'https://www.gbif.org/occurrence/{result["gbifId"]}',
```

Comment:
Use dot notation and is it gbifId or gbifID?

Reply:
we set this field to "gbifId" in the run_api_call() function, but changed it to gbifID to be consistent with the gbif API and ods

Reply:
In that case it would be most logical if the gbifId in the result should also be gbifID? Do you see what I am trying to point out with these IDs :)

Reply:
Yes, I changed it to gbifID to be consistent